propagation.go

```diff
@@ -178,8 +187,23 @@ func extractTraceContext(opaqueCarrier interface{}) (SpanContext, error) {
 		return spanContext, ot.ErrSpanContextNotFound
 	}
 
-	if !spanContext.Suppressed && (spanContext.SpanID == 0 != (spanContext.TraceIDHi == 0 && spanContext.TraceID == 0)) {
-		return spanContext, ot.ErrSpanContextCorrupted
+	// When the context is not suppressed and one of Instana ID headers set.
+	if !spanContext.Suppressed &&
+		(spanContext.SpanID == 0 != (spanContext.TraceIDHi == 0 && spanContext.TraceID == 0)) {
+		sensor.logger.Debug("broken Instana trace context:",
+			" SpanID=", FormatID(spanContext.SpanID),
+			" TraceID=", FormatLongID(spanContext.TraceIDHi, spanContext.TraceID))
+
+		// Check if w3 context was found or not
+		if spanContext.W3CContext.IsZero() {
+			return spanContext, ot.ErrSpanContextCorrupted
+		} else {
```

Comment:
Also I'd rather flip the condition and return the `SpanContext{ W3CContext: spanContext.W3CContext }` in case `spanContext.W3CContext` is not zero, and an error by default, just for readability reasons.

Reply:
👍